frontend/src/features/permits/components/dialog/ApplicationPendingPermitsModalDialog.tsx

```diff
@@ -0,0 +1,127 @@
+import { faExclamationTriangle } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { Dialog, Button, Typography } from "@mui/material";
+import "./ApplicationPendingPermitsModalDialog.scss";
+import { InfoBcGovBanner } from "../../../../common/components/banners/InfoBcGovBanner";
+import { MRT_ColumnDef, MaterialReactTable, useMaterialReactTable } from "material-react-table";
+import { NoRecordsFound } from "../../../../common/components/table/NoRecordsFound";
+import { useContext, useMemo } from "react";
+import { ApplicationListItem } from "../../types/application";
+import { ApplicationPendingPermitsModalColumnDefinition } from "./ApplicationPendingPermitsModalColumnDefinition";
+import { getDefaultNullableVal } from "../../../../common/helpers/util";
+import OnRouteBCContext from "../../../../common/authentication/OnRouteBCContext";
+import { defaultTableInitialStateOptions, defaultTableOptions } from "../../../../common/helpers/tableHelper";
+import { useApplicationsInProgressQuery } from "../../hooks/hooks";
+import { PPC_EMAIL, TOLL_FREE_NUMBER } from "../../../../common/constants/constants";
+
+/**
+ * Dynamically set the column
+ * @returns An array of column headers/accessor keys for Material React Table
+ */
+const getColumns = (): MRT_ColumnDef<ApplicationListItem>[] => {
+    return ApplicationPendingPermitsModalColumnDefinition();
+  };
+
+export const ApplicationPendingPermitsModalDialog = ({
+    showModal,
+    onCancel,
+  }: {
+    showModal: boolean;
+    onCancel: () => void;
+  }) => {
+    
+    const applicationPermitsPendingQuery = useApplicationsInProgressQuery({
+        pendingPermits: true,
+      });
```

Comment:
Do we really need to call this query here (again, since it's already called in ApplicationsInProgressList.tsx below, alongside the normal applications in progress query)? I feel like we do need the two queries alongside each other inside ApplicationsInProgressList (since the warning needs to be displayed somehow before opening the modal), but we don't need it here. Instead, pass the query object props (data, pagination/filter/sorting fields etc) or whatever's needed here into this component as props.

Reply:
In general, I'm usually skeptical of using queries/mutations inside modals/dialogs, and feel like those actions (with large side effects) should be done outside of modal components (preferably inside page components) and useful props (including actions/methods) can be passed into the modal/dialog as props so they can be used.

Reply:
depends if pagination is required here or not, which I'll clarify. That initial query *should* be cached, so I know what you're saying, but we shouldn't be paying the penalty of a double hit to the server... but I'll still keep an eye on that during local testing.

Reply:
pagination is required - so updated this dialog to make use of our pagination patterns

Reply:
I'm not talking about penalties of calling the API twice, I'm just saying in general it's not a good practice, unless there's a compelling reason to do so. In this case, the query has already been done inside ApplicationsInProgressList, and the dialog is simply responsible for displaying it (otherwise we'd be displaying it alongside the actual AIP table which wouldn't look good visually), so you can simply pass the query data, meta totalItems, and whatever's needed from the query into the Dialog component as props without having to call the query again inside the Dialog. This doesn't really have anything to do with caching. 

Regarding the pagination scenario, we can create a custom hook (call it something like usePendingPermits) that contains the sorting and pagination state alongside the pending query, and call it inside the AIPList (replacing the current pending query), and pass the values returned from this hook as props into the Dialog.